wordle-app/pages/WordOfTheDayGame.vue

```diff
@@ -0,0 +1,290 @@
+        </v-col>
+      </v-row>
+    </v-card>
+
+    <UserNameDialog @okay="nameDialogClosed()"  v-model:show="showUserNameDialog" v-model:userName="nameUserNameDialog" />
+    
+  </v-container>
+</template>
+
+<script setup lang="ts">
+import { Game, GameState } from "../scripts/game";
+import nuxtStorage from "nuxt-storage";
+import Axios from 'axios'
+import type { Player } from "../scripts/player";
+import UserNameDialog from "~/components/UserNameDialog.vue";
+
+const router = useRouter();
+const game = reactive(new Game());
+game.startNewGame();
+provide("GAME", game);
+
+// User name dialog
+const showUserNameDialog = ref(false);
+const nameUserNameDialog = ref("");
+
+function nameDialogClosed() {
+  showUserNameDialog.value = false;
+  startStopwatch();
+}
+
+var showWordsList = ref(false);
+
+onMounted(() => {
+
+  window.addEventListener("keyup", onKeyup);
+
+
+  // Assignment 3 Task 2
+  // Check if user name is stored in local storage
+  checkUserNameLocalStorage();
+  checkUserName();
+
+  if(!showUserNameDialog.value){
+    startStopwatch();
+  }
+
+});
+
+
+onUnmounted(() => {
+  window.removeEventListener("keyup", onKeyup);
+});
+
+async function getWordFromApi(): Promise<string> {
+  let wordUrl = "word/wordOfTheDay";
+  const response = await Axios.get(wordUrl);
+  return response.data;
+}
+
+function onKeyup(event: KeyboardEvent) {
+  if (showUserNameDialog.value) {
+    if (event.key === 'Enter') {
+      showUserNameDialog.value = false;
+      nuxtStorage.localStorage.setData("userName", nameUserNameDialog.value);
+    }
+  } else {
+    if (event.key === "Enter") {
+      game.submitGuess();
+    } else if (event.key == "Backspace") {
+      game.removeLastLetter();
+    } else if (event.key.match(/[A-z]/) && event.key.length === 1) {
+      game.addLetter(event.key.toUpperCase());
+    }
+  }
+
+}
+
+async function checkUserNameLocalStorage() {
+  
+  if(nuxtStorage.localStorage.getData("userName") != null){
+    nameUserNameDialog.value = nuxtStorage.localStorage.getData("userName");
+    showUserNameDialog.value = false;
+  }
+  else {
+    nameUserNameDialog.value = "Guest";
+    showUserNameDialog.value = true;
+  }
+
+}
+
+async function checkUserName() {
+  const response = await Axios.get("/Player/Player?playerName=" + nameUserNameDialog.value);
+  const player: Player = response.data;
+
+  if(player.name == "Guest"){
+    showUserNameDialog.value = true;
+  }
+}
+
+function postScore() {
+    console.log("PostScoreEntered")
+    //console.log("Stopwatch seconds: " + stopwatch.seconds.value)
+    var attempts = 0;
+    if(game.gameState == GameState.Won){
+      attempts = game.guessIndex + 1;
+    }else{
+      attempts = game.guesses.length + 5;
+    }
+    Axios.post("Player/AddPlayer", {
+      Name: nameUserNameDialog.value,
+      GameCount: 1,
+      AverageAttempts: attempts,
+      AverageSecondsPerGame: seconds.value,
+    })
```

Comment:
```suggestion
    Axios.post("Player/AddPlayer", {
      Name: nameUserNameDialog.value,
      GameCount: 1,
      AverageAttempts: game.guessIndex + 1;,
      AverageSecondsPerGame: seconds.value,
    })
```

guessIndex should always reflect your attempts - 1 so if you've lost the guessIndex will be at 5. If you one then it will also be correct.